src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1023,33 +1023,36 @@ namespace AirLoopHVACDOAS {
         int constexpr winterDesignDayTypeIndex(10);
 
         for (size_t i = 1; i <= state.dataWeatherManager->DesDayInput.size(); i++) {
+            auto &desDayInput = state.dataWeatherManager->DesDayInput(i);
             // Summer design day
-            if (state.dataWeatherManager->DesDayInput(i).DayType == summerDesignDayTypeIndex && state.dataAirLoopHVACDOAS->SummerDesignDayFlag) {
-                this->SizingCoolOATemp = state.dataWeatherManager->DesDayInput(i).MaxDryBulb;
-                if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::WetBulb) { // wet bulb temperature
-                    this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdbTwbPb(
-                        state, this->SizingCoolOATemp, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::DewPoint) { // dewpoint
-                    this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(
-                        state, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::HumRatio) {
-                    this->SizingCoolOAHumRat = state.dataWeatherManager->DesDayInput(i).HumIndValue;
-                } // else { // What about other cases?
-                state.dataAirLoopHVACDOAS->SummerDesignDayFlag = false;
+            if (desDayInput.DayType == summerDesignDayTypeIndex) {
```

Comment:
I'm not a fan of relying on the design day type index (can just take the make of all design days for cooling and the min ODB of all design days for heating), nor of pulling from design day inputs, but that was the method before, so it can stand for this round.

Reply:
All these review comments are valid. When making these changes I realized that SizingPeriod:WeatherFileDays were not included here and (after making these changes) created new variables to be set during the pass through the sizing periods during sizing. That works good but is very different from what is done here so I figured to get this fixed for the next release and post the new changes for further discussion. Since all sizing days are processed through the weather manager I figured I could capture the data there and not have to process raw SizingPeriod inputs. This of course could probably be improved so I was not able to get this ready by the next release.

        if (state.dataGlobal->CurrentTime == state.dataGlobal->TimeStepZone &&
                (state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).KindOfEnvrn ==
                    DataGlobalConstants::KindOfSim::RunPeriodDesign ||
            state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay)) {
            for (int i = 0; i < state.dataWeatherManager->TodayOutDryBulbTemp.size(); ++i) {
                if (state.dataWeatherManager->TodayOutDryBulbTemp[i] >
                    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOAT) {
                    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOAT =
                        state.dataWeatherManager->TodayOutDryBulbTemp[i];
                    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOADP =
                        state.dataWeatherManager->TodayOutDewPointTemp[i];
                }
                if (state.dataWeatherManager->TodayOutDryBulbTemp[i] <
                    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOAT) {
                    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOAT =
                        state.dataWeatherManager->TodayOutDryBulbTemp[i];
                    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOADP =
                        state.dataWeatherManager->TodayOutDewPointTemp[i];
                }
            }
        }